.github/actions/run-e2e-leg/action.yaml

```diff
@@ -0,0 +1,186 @@
+name: 'Run an e2e leg'
+description: 'Will run all tests under one e2e leg with specific configurations'
+inputs:
+  vlogger-image:
+    description: 'Name of the vertica logger image'
+    required: true
+  operator-image:
+    description: 'Name of the operator image'
+    required: true
+  vertica-image:
+    description: 'Name of the vertica server image'
+    required: true
+  artifact-suffix:
+    description: 'Version of the suffix artifact file'
+    required: false
+    default: ''
+  vertica-deployment-method:
+    description: 'Vertica deployment method (admintools, vclusterops)'
+    required: true
+  vertica-superuser-name:
+    description: 'Vertica superuser name'
+    required: false
+    default: dbadmin
+  # secret not supported in GitHub action, treat this as input here
+  dockerhub-username:
+    description: 'Docker Hub username'
+    required: true
+  # secret not supported in GitHub action, treat this as input here
+  dockerhub-token:
+    description: 'Docker Hub token'
+    required: true
+  leg-identifier:
+    description: 'Testing leg identifier'
+    required: true
+  communal-storage-type:
+    description: 'The type of communal storage used (hostpath, s3, azb)'
+    required: false
+    default: 'hostpath'
+  deploy-with:
+    description: 'Operator deployment method (helm, olm)'
+    required: false
+    default: 'helm'
+  e2e-parallelism:
+    description: 'Value of E2E_PARALLELISM'
+    required: false
+    default: ''
+  controllers-scope:
+    description: 'Controllers deployment scope (cluster, namespace)'
+    required: false
+    default: 'cluster'
+  helm-overrides:
+    description: 'Value of HELM_OVERRIDES (to be used in helm install command)'
+    required: false
+    default: ''
+  minimum-vertica-image:
+    description: 'The minimum version of the vertica server image allowed'
+    required: false
+    default: ''
+  need-base-vertica-image:
```

Comment:
This flag is added here to avoid the need to pull and push base vertica image when it is not needed. Previously we always export `BASE_VERTICA_IMG`, which leads to unnecessary image pull and push for the base image, and may also require docker login even if `vertica-image` is not in private repo, in which case the docker login step will actually be skipped because we only check the prefix `docker.io` of `vertica-image`, leading to image pull error.